src/components/AutoCompleteInput/AutoCompleteInput.js

```diff
@@ -1,258 +1,232 @@
-      this.stopTracking();
+      stopTracking();
       return;
     }
 
     const actualToken = lastToken.slice(1);
 
     // if trigger is not configured step out from the function, otherwise proceed
-    if (!currentTrigger) {
+    if (!handleMentionsTrigger) {
       return;
     }
 
-    this.setState({ currentTrigger }, () => {
-      if (!this.isTrackingStarted) this.startTracking();
-    });
+    if (!isTrackingStarted.current) {
+      startTracking('@');
+    }
 
-    this.updateSuggestions(actualToken);
+    updateSuggestions({ query: actualToken, trigger: '@' });
   };
 
-  handleSuggestions = (text) => {
-    // react native is not consistent in order of execution of onSelectionChange and onTextChange
-    // with android and iOS. onSelectionChange gets executed first on iOS (which is ideal for our scenario)
-    // Although on android, this order is reversed. So need to add following 0 timeout to make sure that
-    // onSelectionChange is executed first before we proceed with handleSuggestions.
-    setTimeout(() => {
-      const { selectionEnd: selectionEnd } = this.state;
-
+  const handleSuggestions = (text) => {
+    setTimeout(async () => {
       if (
         text.slice(selectionEnd - 1, selectionEnd) === ' ' &&
-        !this.state.isTrackingStarted
+        !isTrackingStarted.current
       ) {
-        this.stopTracking();
-        return;
+        stopTracking();
+      } else if (!(await handleCommand(text))) {
+        handleMentions({ selectionEnd, text });
       }
-
-      if (this.handleCommand(text)) return;
-
-      this.handleMentions(text);
     }, 100);
   };
 
-  render() {
-    const { t, value } = this.props;
-
-    return (
-      <InputBox
-        ref={this.props.setInputBoxRef}
-        placeholder={t('Write your message')}
-        onChangeText={(text) => {
-          this.handleChange(text);
-        }}
-        value={value}
-        onSelectionChange={this.handleSelectionChange}
-        multiline
-        {...this.props.additionalTextInputProps}
-      />
-    );
-  }
-}
-
-export default withTranslationContext(AutoCompleteInput);
+  return (
+    <InputBox
+      multiline
+      onChangeText={(text) => {
+        handleChange(text);
+      }}
+      onSelectionChange={handleSelectionChange}
+      placeholder={t('Write your message')}
+      ref={setInputBoxRef}
+      testID='auto-complete-text-input'
+      value={value}
+      {...additionalTextInputProps}
+    />
+  );
+};
+
+AutoCompleteInput.propTypes = {
+  /**
+   * Additional props for underlying TextInput component. These props will be forwarded as it is to TextInput component.
+   *
+   * @see See https://facebook.github.io/react-native/docs/textinput#reference
+   */
+  additionalTextInputProps: PropTypes.object,
+  /**
+   * @param text string
+   */
+  onChange: PropTypes.func,
+  setInputBoxRef: PropTypes.func,
+  triggerSettings: PropTypes.object,
+  value: PropTypes.string,
```

Comment:
after our chat, prob wanna add descriptions to these guys

Reply:
nah chill bro

Reply:
I'll handle it within my MessageInput changes